sdk/servicebus/Azure.Messaging.ServiceBus/samples/TopicFilters/Program.cs

```diff
@@ -0,0 +1,242 @@
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+
+namespace TopicSubscriptionWithRuleOperationsSample
+{
+    public class Program
+    {
+        private const string TopicName = "TopicSubscriptionWithRuleOperationsSample";
+
+        private const string NoFilterSubscriptionName = "NoFilterSubscription";
+        private const string SqlFilterOnlySubscriptionName = "SqlFilterOnlySubscription";
+        private const string SqlFilterWithActionSubscriptionName = "SqlFilterWithActionSubscription";
+        private const string CorrelationFilterSubscriptionName = "CorrelationFilterSubscription";
+
+        private static ServiceBusClient s_client;
+        private static ServiceBusAdministrationClient s_adminClient;
+        private static ServiceBusSender s_sender;
+
+        public static async Task Main(string[] args)
+        {
+            var command = new RootCommand("Demonstrates the Topic Filters feature of Azure Service Bus.")
+            {
+                new Option<string>(
+                    alias: "--namespace",
+                    description: "Fully qualified Service Bus Queue namespace to use") {Name = "FullyQualifiedNamespace"},
+                new Option<string>(
+                    alias: "--connection-variable",
+                    description: "The name of an environment variable containing the connection string to use.") {Name = "Connection"},
+            };
+            command.Handler = CommandHandler.Create<string, string>(RunAsync);
+            await command.InvokeAsync(args);
+        }
+
+        private static async Task RunAsync(string fullyQualifiedNamespace, string connection)
+        {
+            if (!string.IsNullOrEmpty(connection))
+            {
+                s_client = new ServiceBusClient(Environment.GetEnvironmentVariable(connection));
+                s_adminClient = new ServiceBusAdministrationClient(Environment.GetEnvironmentVariable(connection));
+            }
+            else if (!string.IsNullOrEmpty(fullyQualifiedNamespace))
+            {
+                var defaultAzureCredential = new DefaultAzureCredential();
+                s_client = new ServiceBusClient(fullyQualifiedNamespace, defaultAzureCredential);
+                s_adminClient = new ServiceBusAdministrationClient(fullyQualifiedNamespace, defaultAzureCredential);
+            }
+            else
+            {
+                throw new ArgumentException(
+                    "Either a fully qualified namespace or a connection string environment variable must be specified.");
+            }
+
+            Console.WriteLine($"Creating topic {TopicName}");
+            await s_adminClient.CreateTopicAsync(TopicName);
+
+            s_sender = s_client.CreateSender(TopicName);
+
+            // First Subscription is already created with default rule. Leave as is.
```

Comment:
Should we consider defining what a filter is and why we'd want to use them?   Alternative:  point at existing documentation with a (very) brief description